### PR TITLE
[Merged by Bors] - Reduce validator monitor logging verbosity

### LINKED ISSUE #2541: Offer an option to make validator monitor logs less verbose
## Description

This is not a bug report, but a feature request.

When a beacon chain handles hundreds of validators, the info-level logs become a bit too verbose and the log size increases proportionally with the number of validators. If these logs are aggregated by cloud platforms (AWS and Google Cloud) for monitoring and future analysis, the verbosity translates into a bigger bill.

It will be great if we can either 1) turn off INFO logs and only collect more critical log entries (but please output something per epoch to make sure we know the node is still live and functional), or 2) aggregate information for all validators into a single log entry per epoch, probably with _avg_, _max_, _min_, and _count_ values of some metrics, or 3) sample logs somehow, like 1 out of N.

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

Resolves #2541

## Proposed Changes

Reduces verbosity of validator monitor per epoch logging by batching info logs for multiple validators.

Instead of a log for every validator managed by the validator monitor, we now batch logs for attestation records for previous epoch.

Before:
```log
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 1, epoch: 65875, matched_head: true, matched_target: true, inclusion_lag: 0 slot(s), service: val_mon
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 2, epoch: 65875, matched_head: true, matched_target: true, inclusion_lag: 0 slot(s), service: val_mon
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 3, epoch: 65875, matched_head: true, matched_target: true, inclusion_lag: 0 slot(s), service: val_mon
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 4, epoch: 65875, matched_head: true, matched_target: true, inclusion_lag: 0 slot(s), service: val_mon
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 5, epoch: 65875, matched_head: false, matched_target: true, inclusion_lag: 0 slot(s), service: val_mon
Sep 20 06:53:08.239 WARN Attestation failed to match head        validator: 5, epoch: 65875, service: val_mon
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 6, epoch: 65875, matched_head: false, matched_target: true, inclusion_lag: 0 slot(s), service: val_mon
Sep 20 06:53:08.239 WARN Attestation failed to match head        validator: 6, epoch: 65875, service: val_mon
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 7, epoch: 65875, matched_head: true, matched_target: false, inclusion_lag: 1 slot(s), service: val_mon
Sep 20 06:53:08.239 WARN Attestation failed to match target      validator: 7, epoch: 65875, service: val_mon
Sep 20 06:53:08.239 WARN Sub-optimal inclusion delay             validator: 7, epoch: 65875, optimal: 1, delay: 2, service: val_mon
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validator: 8, epoch: 65875, matched_head: true, matched_target: false, inclusion_lag: 1 slot(s), service: val_mon
Sep 20 06:53:08.239 WARN Attestation failed to match target      validator: 8, epoch: 65875, service: val_mon
Sep 20 06:53:08.239 WARN Sub-optimal inclusion delay             validator: 8, epoch: 65875, optimal: 1, delay: 2, service: val_mon
Sep 20 06:53:08.239 ERRO Previous epoch attestation missing      validator: 9, epoch: 65875, service: val_mon
Sep 20 06:53:08.239 ERRO Previous epoch attestation missing      validator: 10, epoch: 65875, service: val_mon
```

after
```
Sep 20 06:53:08.239 INFO Previous epoch attestation success      validators: [1,2,3,4,5,6,7,8,9] , epoch: 65875, service: val_mon
Sep 20 06:53:08.239 WARN Previous epoch attestation failed to match head, validators: [5,6], epoch: 65875, service: val_mon
Sep 20 06:53:08.239 WARN Previous epoch attestation failed to match target, validators: [7,8], epoch: 65875, service: val_mon
Sep 20 06:53:08.239 WARN Previous epoch attestations had sub-optimal inclusion delay, validators: [7,8], epoch: 65875, service: val_mon
Sep 20 06:53:08.239 ERRO Previous epoch attestation missing      validators: [9,10], epoch: 65875, service: val_mon
```

The detailed individual logs are downgraded to debug logs.

### CODE REVIEW

File: beacon_node/beacon_chain/src/validator_monitor.rs
Comment (paulhauner) [inline beacon_node/beacon_chain/src/validator_monitor.rs]:
```suggestion
                "Previous epoch attestation(s) had sub-optimal inclusion delay";
```

File: beacon_node/beacon_chain/src/validator_monitor.rs
Comment (paulhauner) [inline beacon_node/beacon_chain/src/validator_monitor.rs]:
```suggestion
                "Previous epoch attestation(s) failed to match target";
```

File: beacon_node/beacon_chain/src/validator_monitor.rs
Comment (paulhauner) [inline beacon_node/beacon_chain/src/validator_monitor.rs]:
```suggestion
                "Previous epoch attestation(s) failed to match head";
```

File: beacon_node/beacon_chain/src/validator_monitor.rs
Comment (paulhauner) [inline beacon_node/beacon_chain/src/validator_monitor.rs]:
```suggestion
                "Previous epoch attestation(s) missing";
```

File: beacon_node/beacon_chain/src/validator_monitor.rs
Comment (paulhauner) [inline beacon_node/beacon_chain/src/validator_monitor.rs]:
```suggestion
                "Previous epoch attestation(s) success";
```

**Review by paulhauner — CHANGES_REQUESTED** [review]
Nice, this should save lots of log lines.

I made a few very minor suggestions. I'm happy to merge after they're applied :)

**Comment by michaelsproul** [comment]
@pawanjay176 I think this should be ready for review again once #2618 is merged and this PR is updated for those changes?

**Review by paulhauner — APPROVED** [review]
Nice one!

bors r+

**Pull request closed without merging**
